src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2432,7 +2461,31 @@ namespace HeatBalanceSurfaceManager {
 										QRadSWwinAbsTot( SurfNum ) += QRadSWwinAbsLayer( Lay, SurfNum );
 									}
 									QRadSWwinAbsTotEnergy( SurfNum ) = QRadSWwinAbsTot( SurfNum ) * TimeStepZoneSec;
-								} // IF (SurfaceWindow(SurfNum)%WindowModelType /= WindowBSDFModel) THEN
+                } else if( inExtWindowModel->isExternalLibraryModel() ) {
+                  std::pair< double, double > incomingAngle = getSunWCEAngles( SurfNum2, BSDFHemisphere::Incoming );
+                  double Theta = incomingAngle.first;
+                  double Phi = incomingAngle.second;
+
+                  std::shared_ptr< CMultiLayerScattered > aLayer =
```

Comment:
Why does this need to be a `std::shared_ptr`? It doesn't seem to actually be "shared" anywhere and just adds overhead.